### ISSUE TITLE
Improve scores from Security Headers and the Qualys SSL Server Test

### ISSUE THREAD

**Issue by Kenneth-Barber** [2factorauth/twofactorauth#3829]
twofactorauth.org scores a D for its use of Security Headers. Please get this to an A+.
https://securityheaders.com/?q=twofactorauth.org&followRedirects=on

Even though twofactorauth.org scores A+ in the Qualys SSL Server Test, there is still room for improvement in the numeric scores. Please drop support for TLS 1.0, TLS 1.1, and weak cipher suites within TLS 1.2.
https://www.ssllabs.com/ssltest/analyze.html?d=twofactorauth.org

**Comment by Carlgo11** [comment]
@DubbySlashy any recommendations on how to do that with gh-pages?

**Comment by Kenneth-Barber** [comment]
@Carlgo11 I don't have any experience with gh-pages.

**Comment by kmpoppe** [comment]
@Carlgo11 I think this will work, but requires work, that I cannot perform within a short amount of time:
https://www.rzegocki.pl/blog/custom-http-headers-with-github-pages/

**Comment by Carlgo11** [comment]
@kmpoppe The pros of running a Heroku app for that would be minimal and the cons of linking to that JS file, from an untrusted source, would outweigh those pros.


@DubbySlashy this has been discussed previously in #1527 & #2995. It's generally good practice to search for previous issues or PRs related to your question before posting it.

We looked into supplying custom headers a few years ago (in #1527) but gh-pages makes it pretty much impossible.
As for the TLS ciphers, it's supplied by Cloudflare and we can't specifically define what ciphers to use. 

I checked the minimum TLS version to TLS 1.1 but I really don't see any reason for setting it higher than that. Any MITM scan would reveal the requested domain name of the request even with TLS and since we're running a static page without any user input the MITM would see exactly the same thing that the user sees. _(That applies to <TLS 1.3, unsure about higher versions)_

That said, we do value security and we've chosen to redirect all eligible requests to the HTTPS version of our site however that's more of a way for us to do our part in transitioning the web over to HTTPS.
